GitCommands/Settings/AppSettings.cs

```diff
@@ -615,7 +615,7 @@ public static bool DontConfirmAmend
 
         public static bool DontConfirmPushNewBranch
         {
-            get { return GetBool("DontConfirmPushNewBranch", false); }
+            get { return GetBool("DontConfirmPushNewBranch", true); }
```

Comment:
The Advanced settings page should contain settings that require advanced knowledge to change the default value of these settings. Such a change is considered risky and should be done with caution.
By no means unchecking the DontConfirmPushNewBranch option can not be considered risky or requiring advanced knowledge.